### DISCUSSION TITLE
Configuring GitHub Codespaces with a Custom Development Container for Team Collaboration

### DISCUSSION THREAD

**Discussion by ariollex1** [community/community#148133] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I want to set up GitHub Codespaces to allow my team to develop in the same environment, but I'm having trouble configuring the development container to include all the required dependencies. How can I configure the .devcontainer file to ensure that everyone gets the correct setup automatically when they open Codespaces?

**Comment by SheatA3Vp** [comment, marked as answer]
To configure a GitHub Codespace with the right development environment, you’ll need to create a _.devcontainer_ folder in your repository, which will contain the necessary configuration for setting up the container. You can define a _devcontainer.json_ file to specify the image, extensions, and tools you want included. Here's a simple example:

```
{
  "name": "Node.js & MongoDB Dev Environment",
  "image": "mcr.microsoft.com/vscode/devcontainers/javascript-node:0-14",
  "features": {
    "mongodb": "latest"
  },
  "extensions": [
    "dbaeumer.vscode-eslint",
    "ms-vscode.vscode-node-azure-pack"
  ]
}
```

This configuration will ensure that when anyone opens the repository in Codespaces, they'll have a Node.js environment with MongoDB and the necessary VS Code extensions automatically installed. If you need to include additional dependencies, you can also add a _Dockerfile_ to further customize the environment.
